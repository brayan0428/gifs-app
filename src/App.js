import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

function App() {
  const [categories, setCategories] = useState([]);

  const addCategory = (category) => {
    setCategories([category, ...categories]);
  };

  return (
    <div className="App">
      <h2>Gifs App</h2>
      <AddCategory addCategory={addCategory} />
      <hr />
      {categories.map((category) => (
        <GifGrid key={category.id} category={category.name} />
      ))}
    </div>
  );
}

export default App;
