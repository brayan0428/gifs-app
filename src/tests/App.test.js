import React from "react";
import { shallow } from "enzyme";
import App from "../App";

describe("<App />", () => {
  test("Validando el componente", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
