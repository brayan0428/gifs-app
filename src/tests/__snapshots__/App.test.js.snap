// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Validando el componente 1`] = `
<div
  className="App"
>
  <h2>
    Gifs App
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
</div>
`;
